### DISCUSSION TITLE
How to recover docker registry images  from directory in case of postgres database failure

### DISCUSSION THREAD

**Discussion by webdevcoach** [goharbor/harbor#18929] [General]
Hello,
My harbor registry failed and my database table is corrupted, is there any way to restore images from directory ?
since database tables are broken i can not pull and push images . is there any way from harbor to read these docker layers and blob files directly into database.

**Comment by zyyw** [comment]
The current workaround might be deploy a new registry/distribution that using the current harbor registry image storage data. And deploying a new harbor instance and replicate the images from the above new registry/distribution to the new harbor instance. But the system settings / project settings that are stored in db will not be able to be recovered.
